new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiConfigMap.java

```diff
@@ -0,0 +1,610 @@
+    DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+    Date date = new Date();
+    final String imageTag = dateFormat.format(date) + "-" + System.currentTimeMillis();
+    // Add repository name in image name for Jenkins runs
+    final String imageName = REPO_USERNAME.equals(REPO_DUMMY_VALUE) ? MII_IMAGE_NAME : REPO_NAME + MII_IMAGE_NAME;
+    final String image = imageName + ":" + imageTag;
+
+    // build the model file list
+    final List<String> modelList = Collections.singletonList(MODEL_DIR + "/" + WDT_MODEL_FILE);
+
+    // build an application archive using what is in resources/apps/APP_NAME
+    assertTrue(buildAppArchive(defaultAppParams()
+        .srcDir(APP_NAME)), String.format("Failed to create app archive for %s", APP_NAME));
+
+    // build the archive list
+    String zipFile = String.format("%s/%s.zip", ARCHIVE_DIR, APP_NAME);
+    final List<String> archiveList = Collections.singletonList(zipFile);
+
+    // Set additional environment variables for WIT
+    checkDirectory(WIT_BUILD_DIR);
+    Map<String, String> env = new HashMap<>();
+    env.put("WLSIMG_BLDDIR", WIT_BUILD_DIR);
+
+    // build an image using WebLogic Image Tool
+    logger.info("Create image {0} using model directory {1}", image, MODEL_DIR);
+    boolean result = createMiiImage(
+        defaultWitParams()
+            .modelImageName(imageName)
+            .modelImageTag(imageTag)
+            .modelFiles(modelList)
+            .modelArchiveFiles(archiveList)
+            .wdtVersion(WDT_VERSION)
+            .env(env)
+            .redirect(true));
+
+    assertTrue(result, String.format("Failed to create the image %s using WebLogic Image Tool", image));
+
+    /* Check image exists using docker images | grep image tag.
+     * Tag name is unique as it contains date and timestamp.
+     * This is a workaround for the issue on Jenkins machine
+     * as docker images imagename:imagetag is not working and
+     * the test fails even though the image exists.
+     */
+    assertTrue(doesImageExist(imageTag),
+        String.format("Image %s doesn't exist", image));
+
+    return image;
+  }
+
+  private void createRepoSecret(String domNamespace) throws ApiException {
+    V1Secret repoSecret = new V1Secret()
+            .metadata(new V1ObjectMeta()
+                    .name(REPO_SECRET_NAME)
+                    .namespace(domNamespace))
+            .type("kubernetes.io/dockerconfigjson")
+            .putDataItem(".dockerconfigjson", dockerConfigJson.getBytes());
+
+    boolean secretCreated = false;
+    try {
+      secretCreated = createSecret(repoSecret);
+    } catch (ApiException e) {
+      logger.info("Exception when calling CoreV1Api#createNamespacedSecret");
+      logger.info("Status code: " + e.getCode());
+      logger.info("Reason: " + e.getResponseBody());
+      logger.info("Response headers: " + e.getResponseHeaders());
+      //409 means that the secret already exists - it is not an error, so can proceed
```

Comment:
409 means "conflict".  A change has been made since the last time you read the resource.  It's an optimistic locking  failure.